### ISSUE TITLE
Linter error is not appearing in the gutter in visual mode

### ISSUE THREAD

**Issue by astayleraz** [rstudio/rstudio#10949]
### System details

    RStudio Edition : 
    RStudio Version : 2022.06.0-daily+284
    OS Version      : ubuntu 20
    R Version       : 4.1.2

Failing automation test: https://build.rstudioservices.com/job/IDE/job/qa-opensource-automation/1247/testReport/rstudio_server_pro.tests.test_VisualMarkdownEditor/TestVisualMarkdownEditor/test_linting_error_appears_in_gutter/

Note: Due to the number of automation failures we are having I was not able to narrow down the exact build this started to fail but it was working previously in SW.

### Steps to reproduce the problem

1. Create a rmarkdown file via console command `.rs.api.executeCommand('newRMarkdownDoc')` and choose to create a empty document.
2. Install the lintr package via the console. `install.packages("lintr")`
3. Restart r via console command `.rs.api.executeCommand('restartR')`
4. Insert the following in the source file:
~~~
```{r cars}
summary(cars,
         planes
         trains
         )
```
~~~
5. Click to view in visual mode.
6. Save the file.

### Describe the problem in detail
The linter error is not appearing in the gutter in visual mode. It shows as the image below:

![Screen Shot 2022-04-11 at 10 07 27 AM](https://user-images.githubusercontent.com/1482677/162799108-7dd0fb7d-af17-4e16-8933-63089f8de3c1.png)

### Describe the behavior you expected
The test is expecting to see the linter error below:
<img width="277" alt="Screen Shot 2022-04-11 at 10 46 21 AM" src="https://user-images.githubusercontent.com/1482677/162798942-143f28c8-8e40-40bc-9750-428dc9dd5dc5.png">

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Do you mean the error message tooltip?

In your first screenshot, the cursor is not on the line with the associated error, which would explain its absence.

**Comment by astayleraz** [comment]
Yeah, the tooltip error. I do not see it in visual mode and that was the screen shop for the test but even when I move the cursor on the line I do not see the error. Shows as this:
![Screen Shot 2022-04-11 at 11 34 46 AM](https://user-images.githubusercontent.com/1482677/162806631-0621512c-3c54-481e-a4b5-a05740e1cd08.png)

**Comment by rich-rstudio** [comment]
in order for the error to show, you have to resave the file in visual mode, then the linter works

**Comment by mikebessuille** [comment]
Tim, is this related to the gutter stuff you did?

**Comment by timtmok** [comment]
I'm not sure if this was related to the custom marker changes as the behaviour was a little inconsistent before those changes. The events show that the document is linted and returns the correct lint items but they don't update in visual mode. Refreshing the Code Diagnostics seems to partially update the markers as well.

This might be an issue with the callback when linting completes. It should trigger the visual code chunks to update.

**Comment by timtmok** [comment]
The source of the error is how the visual mode chunk was created. The automated test and reproduction example creates the R code at the beginning of the document. When the chunk is created, it doesn't set a scope due to a bug in checking the chunk's index location in the document. This would only occur for the visual mode code chunk at the beginning of a document. The remaining code chunks would get the markers once they load in.

For some reason it works when saving the file but that's due to a different code path because it has to lint the document. The issue would occur when initially switching the document to visual mode and the lint items already exist from a previous run.

**Comment by jgutman** [comment]
Following the reprex above, this appears to be fixed:

<img width="747" alt="image" src="https://user-images.githubusercontent.com/7817881/170087105-a818550d-13ea-4a3d-ba69-5dd7cebf6b75.png">

<img width="381" alt="image" src="https://user-images.githubusercontent.com/7817881/170087175-a264e276-6032-4e40-ab69-f87e0abceb16.png">

in the latest RStudio Server. 

@astayleraz -- could you please check if the automated test you linked is now passing, and re-enable that test?

https://github.com/rstudio/rstudio-ide-automation/blob/0f8c9259918be68b3dd537e2c6da9fc1885439b6/rstudio_server_pro/tests/test_VisualMarkdownEditor.py#L560

**Comment by astayleraz** [comment]
### Verified Fixed
    RStudio Edition    :  Workbench
    RStudio Version(s) : 2022.06.0-daily+440.pro1
    OS Version         : Ubuntu20

### Automated tests 
The automation test on this is now passing. Re-enabling the test and closing this as fixed.
